pkg/controller/controlplane/actuator.go

```diff
@@ -101,6 +101,16 @@ func (a *actuator) Delete(
 	return nil
 }
 
+// ForceDelete forcefully deletes the controlplane.
+func (a *actuator) ForceDelete(
+	ctx context.Context,
+	log logr.Logger,
+	cp *extensionsv1alpha1.ControlPlane,
+	cluster *extensionscontroller.Cluster,
+) error {
+	return a.Delete(ctx, log, cp, cluster)
```

Comment:
WDYT about instead of calling the full `a.Delete()` function here to instead only call 
```go
        // Call Delete on the composed Actuator
	if err := a.Actuator.Delete(ctx, log, cp, cluster); err != nil {
		return err
	}
	if cp.Spec.Purpose == nil || *cp.Spec.Purpose == extensionsv1alpha1.Normal {
		// Delete all remaining remedy controller resources
		return a.forceDeleteRemedyControllerResources(ctx, log, cp)
	}
```
so that we avoid having to wait for the graceful deletion of the `PublicIPAddresses`?

Reply:
+1